app/models/spree/user_decorator.rb

```diff
@@ -8,6 +8,14 @@ def supplier?
     supplier.present?
   end
 
+  def supplier_admin?
+    supplier? && has_admin_role?
+  end
+
+  def market_maker?
```

Comment:
to be clear, a user cannot be both a `market_maker` and a `supplier_admin`?
(totally fine if not, just trying to be clear)